blog-posts/en/angular-http-only-cookie.md

```diff
@@ -0,0 +1,423 @@
+# How to enable HttpOnly cookies in ASP.NET Zero Angular
+
+HttpOnly cookies are a type of cookie that can only be accessed and manipulated by the server through HTTP requests, not by JavaScript or client-side code. They enhance security by preventing cross-site scripting (XSS) attacks that could steal sensitive information stored in cookies. 
+
+First of all, your backend project and your Angular project must be located in the same domain. For example, if your Angular project is located in `https://localhost:44300`. Then your backend project `*.Host` must be located in `https://localhost:44300/api`.
+
+## Step 1: Configure your `*Web.Core` project
+
+### Add HttpOnly cookies to `Authenticate` method
+
+Find the `Authenticate` method and add the above code after the `var accessToken = CreateAccessToken( ...` line. And also you need to add refresh token cookie as well. Your code should look like this:
+
+```csharp
+// Other codes ...
+
+    var accessToken = CreateAccessToken(
+        await CreateJwtClaims(
+            loginResult.Identity,
+            loginResult.User,
+            refreshTokenKey: refreshToken.key
+        )
+    );
+
+    Response.Cookies.Append("Abp.AuthRefreshToken",
+        refreshToken.token,
+        new CookieOptions
+        {
+            Expires = Clock.Now.Add(_configuration.RefreshTokenExpiration),
+            HttpOnly = true,
+            Secure = true,
+            SameSite = SameSiteMode.None
+        }
+    );
+
+    Response.Cookies.Append("Abp.AuthToken",
+        accessToken,
+        new CookieOptions
+        {
+            Expires = Clock.Now.Add(_configuration.AccessTokenExpiration),
+            HttpOnly = true,
+            Secure = true,
+            SameSite = SameSiteMode.None
+        }
+    );
+
+    return new AuthenticateResultModel
+    {
+        AccessToken = accessToken,
```

Comment:
While changing the cookies so that they can be HttpOnly is good, it doesn't solve the problem where the client code has access to the authentication tokens. The tokens are still returned within the auth response bodies to the client (as shown here). The token data would need to be removed from the response bodies to prevent the client from getting access to the auth/refresh tokens.

Reply:
Doing this makes the applicability of the solution really difficult. But if you want, you can remove the return methods and make your application better. This blog post is designed to give you a base for http only tokens.

Reply:
Thanks. Just wanted to make sure that you were aware of the issue.